openslides/utils/autoupdate.py

```diff
@@ -95,9 +97,16 @@ def send_data(message):
             'action': 'deleted' if message['is_deleted'] else 'changed'}
         if not message['is_deleted']:
             data = access_permissions.get_restricted_data(full_data, user)
+
             if data is None:
-                # There are no data for the user so he can't see the object. Skip him.
-                continue
+                # send data if user can see the projector and data is projected
```

Comment:
No. get_all_requirements fetches all requirements from all currently active projector elements. But then you have to go through these requirements and decide whether the current data match one of these. See utils/rest_api.py:127. The method is_currently_required does not work because it expects a view instance. and not a model instance.

When I created this feature/system there was no need to restrict autoupdate data separatly because we used to send also autoupdate date through an internal GET request. But now we don't so we have to refactor the whole ProjectorRequirements system.